src/System.Private.CoreLib/shared/System/Number.NumberBuffer.cs

```diff
@@ -9,26 +9,37 @@ namespace System
 {
     internal static partial class Number
     {
-        private const int NumberMaxDigits = 50;
-
-        private const double Log10V2 = 0.30102999566398119521373889472449;
-
-        // DriftFactor = 1 - Log10V2 - epsilon (a small number account for drift of floating point multiplication)
-        private const double DriftFactor = 0.69;
+        //  We need 1 additional byte, per length, for the terminating null
+        private const int DecimalNumberBufferLength = 50 + 1;
```

Comment:
I kept `DecimalNumberBufferLength` at the current value (`50 + 1`). However, I believe we can make it `30` (29 digits + 1 for rounding).

Reply:
If the rounding digit is 5 then up to 20 additional digits are checked in `NumberBufferToDecimal` (so 50 in total). The number 20 looks arbitrary, theoretically it would be correct to check the entire string...

Reply:
@pentp, maybe I am misremembering, but I thought `decimal` only supported 29 digits max. That is, it has 96-bits for the mantissa and it uses 8-bits for the exponent, which is strictly restricted to a value between 0 and 28, inclusive.

This should mean that you need to consider 29 digits, plus 1 for rounding. Anything more than that shouldn't be impactful.

Reply:
It supports only 29 digits, but if the rounding digit is 5 and the number is even, then it checks the following digits to decide it it should rounding up or down.

Reply:
Yes, that's correct.